src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -0,0 +1,279 @@
+        for (const auto& version : versions)
+        {
+            fx_ver_t ver(-1, -1, -1);
+            if (fx_ver_t::parse(version, &ver, true))
+            {
+                max_ver = max(ver, max_ver);
+            }
+        }
+        pal::string_t max_ver_str = max_ver.as_str();
+        append_path(&sdk_path, max_ver_str.c_str());
+        if (pal::directory_exists(sdk_path))
+        {
+            retval = sdk_path;
+        }
+    }
+    cli_sdk->assign(retval);
+    return !retval.empty();
+}
+
+/* static */
+int fx_muxer_t::execute(const int argc, const pal::char_t* argv[])
+{
+    pal::string_t own_path;
+
+    // Get the full name of the application
+    if (!pal::get_own_executable_path(&own_path) || !pal::realpath(&own_path))
+    {
+        trace::error(_X("Failed to locate current executable"));
+        return StatusCode::LibHostCurExeFindFailure;
+    }
+
+    auto own_dir = get_directory(own_path);
+
+    if (argc <= 1)
+    {
+        return StatusCode::InvalidArgFailure;
+    }
+    if (ends_with(argv[1], _X(".dll"), false))
+    {
+        pal::string_t app_path = argv[1];
+
+        if (!pal::realpath(&app_path))
+        {
+            return StatusCode::LibHostExecModeFailure;
+        }
+
+        runtime_config_t config(get_runtime_config_json(app_path));
+        if (config.get_portable())
+        {
+            pal::string_t fx_dir = resolve_fx_dir(own_dir, &config, app_path);
+            corehost_init_t init(_X(""), _X(""), fx_dir, &config);
+            return policy_load_t::execute_app(fx_dir, &init, argc, argv);
+        }
+        else
+        {
+            corehost_init_t init(_X(""), _X(""), _X(""), &config);
+            return policy_load_t::execute_app(get_directory(app_path), &init, argc, argv);
+        }
+    }
+    else
+    {
+        if (pal::strcasecmp(_X("exec"), argv[1]) == 0)
+        {
+            std::vector<pal::string_t> known_opts = { _X("--depsfile"), _X("--additionalprobingpath") };
+
+            int num_args = 0;
+            std::unordered_map<pal::string_t, pal::string_t> opts;
+            if (!parse_known_args(argc - 2, &argv[2], known_opts, &opts, &num_args))
+            {
+                return InvalidArgFailure;
+            }
+            int cur_i = 2 + num_args;
+            if (cur_i >= argc)
+            {
+                return InvalidArgFailure;
+            }
+
+            // Transform dotnet exec [--additionalprobingpath path] [--depsfile file] dll [args] -> dotnet dll [args]
+
+            std::vector<const pal::char_t*> new_argv(argc - cur_i + 1); // +1 for dotnet
+            memcpy(new_argv.data() + 1, argv + cur_i, (argc - cur_i) * sizeof(pal::char_t*));
+            new_argv[0] = argv[0];
+
+            pal::string_t deps_file = opts.count(_X("--depsfile")) ? opts[_X("--depsfile")] : _X("");
+            pal::string_t probe_path = opts.count(_X("--additionalprobingpath")) ? opts[_X("--additionalprobingpath")] : _X("");
```

Comment:
@brthor Are you planning to pass a hint that muxer can use for non-published apps to continue to support "dotnet <path to app.dll>"?